### PR TITLE
Change from go deps to go modules and fix errors of the migration

### PR DESCRIPTION
**Pull request by arthur29**
Why
======

The Khan is using the legacy `go deps` dependency management tool. This PR proposes to update it to use `go modules`. Some changes were needed to keep the project building correctly.

What was done
======
* Add `go.mod` and `go.sum`
* Remove `Gopkg.toml` and `Gopkg.lock`
* Create some singletons to mongo and elasticsearch clients
* Apply changes to Makefile

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,49 @@
+module github.com/topfreegames/khan
+
+go 1.15
```
Comment (leonardo2204) [inline go.mod:3]: Is this the correct go version?

**Comment by leonardo2204** [comment]
Just a nitpick here, you should avoid doing 2 stuff in the same PR (like changing deps mgmt + refactoring the code). Doing so may bloat the PR size and requires a mindset change when reviewing the PR :)

File: db/migrations.go [hunk 2]
```diff
@@ -1,491 +1,203 @@
...
-// migrations/20160708161944_CreateMembershipApproverField.sql
-// migrations/20160708192007_CreateOwnerIndex.sql
-// migrations/20160713185332_CreateGameCooldownAfterDenyAndDelete.sql
-// migrations/20160713191703_CreateMembershipDenierField.sql
-// migrations/20160728180524_CreateMaxPendingInvitesField.sql
-// migrations/20160728195902_CreateMembershipMessageField.sql
-// migrations/20160729184159_CreateCooldownAfterInviteField.sql
-// migrations/20160819145352_CreateHookTriggerFieldsMetadata.sql
-// migrations/20180517112014_ChangeIDSequenceType.sql
-// DO NOT EDIT!
```
[the remark this answers is not in the document]
Comment (arthur29) [reply db/migrations.go:21]: Yes, this was auto-generated by `go-bindata`

File: go.mod (hunk 1 above)
Comment (arthur29) [reply go.mod:3]: No, this the version that I currently use local, I will fix this

**Pull request closed without merging**